sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveInspectors.scala

```diff
@@ -246,6 +246,9 @@ private[hive] trait HiveInspectors {
    * Wraps with Hive types based on object inspector.
    */
   protected def wrapperFor(oi: ObjectInspector, dataType: DataType): Any => Any = oi match {
+    case _ if dataType.isInstanceOf[UserDefinedType[_]] =>
```

Comment:
>This codepath is shared by many things apart from ORC. Won't those be affected ?

It seems this path is being used in `hiveUDFs.scala` and `hiveWriterContainers.scala`.
Actually, it'd be fine that a value converter for UDT uses the equivalent type (inner sql type) converter. It is a common pattern for other data sources as well. 

>I would put this case in the every end. The reason being UserDefinedType are not that common compared to other types (esp. primitive types). So putting it below in the switch case will be better for perf.

Cool :)